docs: PEM-6014 (#4216)
spectrocloud/librarium: docs/docs-content/profiles/cluster-profiles/modify-cluster-profiles/update-cluster-profile.md

   :::

   ![A view of the cluster profile update widget displaying a new packs versions but changes that require the user's input.](/profiles_cluster-profiles_modify-cluster-profiles_new-version-notifcation-changes-required.webp)

6. Click on the **Apply Changes** button to apply the updates to the profile. If have changes that require your
   attention then click on the **Review changes in Editor** button to start the review process.

7. The differential editor will display the changes between the current YAML configuration and new incoming YAML
   changes. The left side of the editor displays the current configuration. The right side displays the new pack
   version's incoming changes. Review the changes and apply them as needed. Use the three buttons at the bottom to
   navigate through the changes.

   - **Prev**: Click to navigate to the previous change.
   - **Next**: Click to navigate to the next change.
   - **Keep**: Click to apply the current change.
   - **Revert**: Click to revert the accepted change. This button will only appear after you have clicked the **Keep**
     button.

>    The differential editor will display the changes between the current YAML configuration and new incoming YAML changes
>    by highlighting the differences. The color-coded highlights indicate the following:

Reviewer: ```suggestion
   The differential editor will display the changes by highlighting the differences between the configurations. The color-coded highlights indicate the following:
```
maybe? :)

>    - _Yellow highlight_ indicates text that is not present in the new configuration. These may be lines you have added
>      in the current configuration. These lines may also have been removed because they are no longer valid in the new
>      configuration. If you need them, use the **Keep** button to transfer the lines to the new pack version. Otherwise,

Reviewer: ```suggestion
   - _Yellow highlight_ indicates text that is not present in the new configuration. These may be lines you have added
     in the current configuration or lines that have been removed because they are no longer valid in the new
     configuration. If you need them, use the **Keep** button to transfer the lines to the new pack version. Otherwise,
```

- We do not recommend downgrading packs to the previous version.

:::

### Prerequisites

- A cluster profile created in Palette.

- There are updates available for at least one pack in the profile.

- Your Palette account role must have the `clusterProfile.update` permission to update a profile. Refer to the
  [Cluster Profile permissions](../../../user-management/palette-rbac/project-scope-roles-permissions.md#cluster-profile-admin)
  reference for more information about roles and permissions.

### Update Pack Version

1. Log in to [Palette](https://console.spectrocloud.com).

2. From the left **Main Menu**, select **Profiles**.

3. Select the profile you want to update to access the profile details page.

4. In the **Edit Pack** panel, select a specific target version, not a group that ends in `.x`. Palette displays the
   difference between the current version at left and the new version at right. The target version is displayed in the
   header.

   Differences between the displayed configurations are as follows:

   - _Yellow highlight_ indicates text that is not present in the new configuration. These may be lines you have added

>      in the current configuration. These lines may also have been removed because they are no longer valid in the new
>      configuration. If you need them, you can use the **Keep** button to transfer the lines to the new pack version.

Reviewer: ```suggestion
     in the current configuration or lines that have been removed because they are no longer valid in the new
     configuration. If you need them, you can use the **Keep** button to transfer the lines to the new pack version.
```

maybe? :)

     Otherwise, click on **Next** to proceed.

   - _Blue highlight_ indicates additions in the new configuration that are not present in the pack version you are
     using.

     #### Example of Difference Between Current and New Configurations

     ![Screenshot that shows Palette's pack diff user interface with yellow highlight at left and blue highlight at right](/integrations_pack_diffs.webp)

> 5. Check for yellow-highlight in the configuration that are missing in the new configuration. If there are any

Reviewer: ```suggestion
5. Check for yellow-highlights in the current configuration that are missing in the new configuration. If there are any
```

## Apply Profile Updates to Clusters

> You can accept changes to a cluster's cluster profile and override the profile configurations to update a cluster.
> Accepting these change will only apply to the cluster you are updating and will not propagate to the cluster profile or
> other clusters using the same profile.

Reviewer: ```suggestion
You can accept changes to a cluster's cluster profile and override the existing profile configurations to update a cluster.
Accepting these changes will only apply to the cluster you are updating and will not propagate to the cluster profile or
other clusters using the same profile.
```

:::warning

We do not recommend updating a currently deployed cluster profile version to push out changes. Instead, we recommend
creating a _new_ profile version, and then upgrade active clusters to the new version. For information on versioning
profiles, check out [Version a Cluster Profile](version-cluster-profile.md).

:::

### Prerequisites

- A cluster profile created in Palette.

> - A cluster is deployed, healthy, with a cluster profile that has updates available.

Reviewer: ```suggestion
- A deployed and healthy cluster with an associated cluster profile that has  updates available.
```
maybe?
